<launch>
    <!-- Nao ROS drivers -->
    <arg name="nao_ip" default="$(optenv NAO_IP 127.0.0.1)"/>
    <arg name="nao_port" default="$(optenv NAO_PORT 9559)"/>

    <node pkg="diagnostic_aggregator" type="aggregator_node" name="diag_agg" clear_params="true">
        <rosparam command="load" file="$(find nao_bringup)/config/nao_analysers.yaml"/>
    </node>

    <!-- upload nao robot model V40 by default-->
    <include file="$(find nao_description)/launch/robot_state_publisher.launch">
        <arg name="version" value="V40"/>
    </include>

    <!-- publish joint states and basic actuators -->
    <include file="$(find naoqi_driver_py)/launch/naoqi_driver.launch">
        <arg name="nao_ip" value="$(arg nao_ip)"/>
    </include>

    <!-- Blender controllers -->
    <param name="armature_name" value="Armature"/>
    <param name="blender_target_controllers" value="$(find nao_hri)/launch/nao_target_controllers.yaml"/>

    <node pkg="ros_blender_bridge" type="run_target_controllers.py" name="blender_target_controllers" output="screen"/>

    <node pkg="ros_blender_bridge" type="launch_blender" name="launch_blender" output="screen">
        <param name="blend_file" value="$(find nao_hri)/launch/nao.blend"/>
        <param name="use_game_engine" value="false"/>
        <param name="python_script" value="$(find ros_blender_bridge)/scripts/run_blender_node.py"/>
    </node>

    <node pkg="tf" type="static_transform_publisher" name="nao_to_kinect_transform"
          args="-0.045 -0.00 -0.04 0.03 1.1 0.0 1.0 nao_marker base_link 10"/>

    <!-- Nao HRI -->
    <param name="nao_tts_cache" value="$(find nao_hri)/launch/tts_cache.txt"/>
    <node pkg="nao_hri" type="nao_gesture_action_server_node.py" name="nao_gesture_action_server"
          args="--pip=$(arg nao_ip) --pport=$(arg nao_port) --port=0" output="screen"/>
    <node pkg="nao_hri" type="nao_tts_action_server_node.py" name="nao_tts_action_server"
          args="--pip=$(arg nao_ip) --pport=$(arg nao_port) --port=0" output="screen"/>

    <node pkg="nao_hri" type="blender_target_server_node.py" name="nao_gaze_action_server" output="screen">
        <param name="action_server_name" value="gaze"/>
        <param name="controller_name" value="nao_gaze_controller"/>
        <param name="end_effector_frame" value="gaze"/>
        <param name="success_distance" value="0.15"/>
        <param name="axes" value="yz"/>
        <param name="target_name" value="gaze_target"/>
    </node>

    <node pkg="nao_hri" type="blender_target_server_node.py" name="nao_point_left_server" output="screen">
        <param name="action_server_name" value="nao_point_left"/>
        <param name="controller_name" value="nao_larm_controller"/>
        <param name="end_effector_frame" value="l_gripper"/>
        <param name="success_distance" value="0.3"/>
        <param name="axes" value="yz"/>
        <param name="target_name" value="l_gripper_target"/>
    </node>

    <node pkg="nao_hri" type="blender_target_server_node.py" name="nao_point_right_server" output="screen">
        <param name="action_server_name" value="nao_point_right"/>
        <param name="controller_name" value="nao_rarm_controller"/>
        <param name="end_effector_frame" value="r_gripper"/>
        <param name="success_distance" value="0.3"/>
        <param name="axes" value="yz"/>
        <param name="target_name" value="r_gripper_target"/>
    </node>

    <node pkg="hri_framework" type="perception_synthesiser.py" name="perception_synthesiser" output="screen">
        <param name="perception_sources_yaml" value="$(find nao_hri)/launch/perception_sources.yaml"/>
    </node>
</launch>